crates/pathfinder/src/storage/schema/revision_0011.rs

```diff
@@ -21,37 +21,136 @@ pub(crate) fn migrate(transaction: &Transaction) -> anyhow::Result<PostMigration
-                        named_params![
-                            ":idx": idx,
-                            ":transaction_hash": transaction_hash,
-                            ":from_address": &event.from_address.0.as_be_bytes()[..],
-                        ],
-                    )
-                    .context("Insert event data into events table")?;
+                )?;
+
+                stmt.execute(named_params![
+                    ":idx": idx,
+                    ":transaction_hash": transaction_hash,
+                    ":from_address": &event.from_address.0.as_be_bytes()[..],
+                ])
+                .context("Insert event data into events table")?;
 
                 Ok(())
             },
         )?;
+
+        processed_rows += 1;
+
+        if batch_size > 0 && processed_rows % batch_size == 0 {
+            let now = std::time::Instant::now();
+            let total_elapsed = now - start_of_run;
+            let batch_elapsed = now - start_of_batch;
+
+            let total_per_row = total_elapsed.div_f64(processed_rows as f64);
+            let batch_per_row = batch_elapsed.div_f64(batch_size as f64);
+
+            // this is non-scientific, but perhaps the latest helps? seems to be very much off until 75% when divided by 2, 50% when divided by 1.5
+            let est_per_row = (total_per_row + batch_per_row).div_f64(1.5);
+            let remaining = est_per_row * ((todo - processed_rows) as u32);
+
+            println!(
+                "{} rows processed, {:?} elapsed, {:?} per row in batch of {batch_size}, {remaining:?} remaining",
+                processed_rows,
+                total_elapsed,
+                batch_elapsed.div_f64(batch_size as f64),
+            );
+            start_of_batch = now;
+        }
     }
 
-    Ok(PostMigrationAction::None)
+    println!(
+        "decompression time: {decompression_time:?}, parsing_time: {parsing_time:?}, total: {:?}",
+        start_of_run.elapsed()
+    );
+
+    transaction
+        .execute(
+            "CREATE INDEX starknet_events_from_address ON starknet_events(from_address)",
+            [],
+        )
+        .context("Recreate index")?;
+    transaction
+        .execute(
+            "CREATE TRIGGER starknet_events_au
+            AFTER UPDATE ON starknet_events
+            BEGIN
+                INSERT INTO starknet_events_keys(starknet_events_keys, rowid, keys)
+                VALUES (
+                    'delete',
+                    old.rowid,
+                    old.keys
+                );
+                INSERT INTO starknet_events_keys(rowid, keys)
+                VALUES (
+                    new.rowid,
+                    new.keys
+                );
+            END",
+            [],
+        )
+        .context("Recreate trigger")?;
+
+    transaction
+        .execute("DROP INDEX temp_starknet_events_q", [])
+        .context("drop temp index")?;
+
+    anyhow::bail!("roolback!!")
```

Comment:
Unclear if this should be a vacuum? Probably not since its just updates?
```suggestion
    Ok(PostMigrationAction::None)
```

Reply:
I'll try this

Reply:
migrate_db example: `vacuumed in 4024.692105694s, size change: -4457766912, VACUUM returned 0` so not going to put that in.